src/preferences/dialog/dlgprefdeck.cpp

```diff
@@ -178,6 +168,27 @@ DlgPrefDeck::DlgPrefDeck(QWidget* parent,
             this,
             &DlgPrefDeck::slotSetTrackLoadMode);
 
+    comboBoxLoadWhenDeckPlaying->addItem(tr("Allow"), static_cast<int>(LoadWhenDeckPlaying::Allow));
+    comboBoxLoadWhenDeckPlaying->addItem(tr("Allow, but stop deck"), static_cast<int>(LoadWhenDeckPlaying::AllowButStopDeck));
+    comboBoxLoadWhenDeckPlaying->addItem(tr("Reject"), static_cast<int>(LoadWhenDeckPlaying::Reject));
+    int loadWhenDeckPlaying;
+    if (m_pConfig->exists(kConfigKeyLoadWhenDeckPlaying)) {
+        loadWhenDeckPlaying = m_pConfig->getValueString(kConfigKeyLoadWhenDeckPlaying).toInt();
+    } else {
+        // upgrade from older versions
+        if (m_pConfig->getValue(ConfigKey("[Controls]", "AllowTrackLoadToPlayingDeck"), false)) {
+            loadWhenDeckPlaying = static_cast<int>(LoadWhenDeckPlaying::Allow);
+        } else {
+            loadWhenDeckPlaying = static_cast<int>(kDefaultLoadWhenDeckPlaying);
+        }
+    }
+    comboBoxLoadWhenDeckPlaying->setCurrentIndex(comboBoxLoadWhenDeckPlaying->findData(loadWhenDeckPlaying));
+    m_loadWhenDeckPlaying = static_cast<LoadWhenDeckPlaying>(loadWhenDeckPlaying);
+    connect(comboBoxLoadWhenDeckPlaying,
+            QOverload<int>::of(&QComboBox::currentIndexChanged),
+            this,
+            &DlgPrefDeck::slotSetLoadWhenDeckPlaying);
```

Comment:
slotSetLoadWhenDeckPlaying()  fixes the relation of combobox index to value. Lets rename it to 
slotLoadWhenDeckPlayingIndexChanged and do the mapping in source code.

Reply:
Sorry, I don't follow this either.

I just did what the other combo boxes do (e.g. `slotSetTrackLoadMode`). Is there an example of your suggestion? :)

Reply:
It looks like it is sufficient to just rename the function, because it is not a classic setter. It receives the index, fetches the value and than sets the the value. The current name does not expose that an index is expected.